pandas/tests/io/test_pytables.py

```diff
@@ -4999,6 +4999,62 @@ def test_read_nokey_empty(self):
             store.close()
             pytest.raises(ValueError, read_hdf, path)
 
+    # GH10143
+    def test_walk(self):
+        objs = {
+            'df1': pd.DataFrame([1, 2, 3]),
+            'df2': pd.DataFrame([4, 5, 6]),
+            'df3': pd.DataFrame([6, 7, 8]),
+            'df4': pd.DataFrame([9, 10, 11]),
+            's1': pd.Series([10, 9, 8]),
+            'a1': np.array([[1, 2, 3], [4, 5, 6]])
+        }
+
+        with ensure_clean_store('walk_groups.hdf', mode='w') as store:
+            store.put('/first_group/df1', objs['df1'])
+            store.put('/first_group/df2', objs['df2'])
+            store.put('/second_group/df3', objs['df3'])
+            store.put('/second_group/s1', objs['s1'])
+            store.put('/second_group/third_group/df4', objs['df4'])
+            g1 = store._handle.get_node('/first_group')
+            store._handle.create_array(g1, 'a1', objs['a1'])
+
+            expect1 = {
+                '': ({'first_group', 'second_group'}, set()),
+                '/first_group': (set(), {'df1', 'df2'}),
```

Comment:
can you parameterize this test rather than making a long test like this

Reply:
done